lib/internal/source_map/prepare_stack_trace.js

```diff
@@ -49,18 +60,66 @@ const prepareStackTrace = (globalThis, error, trace) => {
         } = sm.findEntry(t.getLineNumber() - 1, t.getColumnNumber() - 1);
         if (originalSource && originalLine !== undefined &&
             originalColumn !== undefined) {
-          str +=
-`\n        -> ${originalSource.replace('file://', '')}:${originalLine + 1}:${originalColumn + 1}`;
+          const originalSourceNoScheme = originalSource.replace('file://', '');
+          if (i === 0) {
+            firstLine = originalLine + 1;
+            firstColumn = originalColumn + 1;
+            firstSource = originalSourceNoScheme;
+          }
+          // Show error in original source context to help user pinpoint it:
+          errorSource = getErrorSource(firstSource, firstLine, firstColumn);
+          // Show both original and transpiled stack trace information:
+          str += `\n        -> ${originalSourceNoScheme}:${originalLine + 1}:` +
+            `${originalColumn + 1}`;
         }
       }
     } catch (err) {
       debug(err.stack);
     }
     return str;
   });
-  return `${errorString}\n    at ${preparedTrace.join('')}`;
+  return `${errorSource}${errorString}\n    at ${preparedTrace.join('')}`;
 };
 
+// Places a snippet of code from where the exception was originally thrown
+// above the stack trace. This logic is modeled after GetErrorSource in
+// node_errors.cc.
+function getErrorSource(firstSource, firstLine, firstColumn) {
+  let exceptionLine = '';
+  let source;
+  try {
+    source = readFileSync(firstSource, 'utf8');
+  } catch (err) {
+    debug(err.stack);
+    return exceptionLine;
+  }
+  const lines = source.split(/(?<=\r?\n)/);
+  const line = lines[firstLine - 1];
+  if (!line) return exceptionLine;
+
+  // Display ^ in appropriate position, regardless of whether tabs or
+  // spaces are used:
+  let prefix = '';
+  for (const character of line.slice(0, firstColumn)) {
+    prefix += (character === '\t') ? '\t' :
+      ' '.repeat(getStringWidth(character));
+  }
+  prefix = prefix.slice(0, -1); // The last character is the '^'.
+
+  exceptionLine = `${firstSource}:${firstLine}\n${line}${prefix}^\n\n`;
+  return exceptionLine;
+}
+
+let getStringWidthCached;
+function getStringWidth(str) {
+  if (!getStringWidthCached && internalBinding('config').hasIntl) {
+    getStringWidthCached = require('internal/util/inspect').getStringWidth;
```

Comment:
The exported `getStringWidth` function has a fallback that works almost like the intl one in case Node.js is not built with intl.